cloud_images/centos7/dcos_vol_setup.sh

```diff
@@ -53,7 +53,12 @@ function main {
     echo "Adding entry to fstab: $fstab"
     echo "$fstab" >> /etc/fstab
     echo -n "Mounting: $device to $mount_location"
-    mount "$mount_location"
+    until grep ^$device /etc/mtab > /dev/null; do sleep 1; echo -n .; mount "$mount_location" || :; done
+    if [ "$mount_location" = "/var/log" ]; then
+      echo "Recreating /var/log/journal"
+      mkdir /var/log/journal
+      systemd-tmpfiles --create --prefix /var/log/journal
+    fi
```

Comment:
So I talked with some colleagues about the issue. We think two things would help the situation:
a) put an explicit `Before=dcos-download.service` so that we can gurantee partitioning happens before our script.

b) create a function ala
```
function noncritical {
  set +e; $*; set -e
}
```
And wrap our echo and sleeps in it like `noncritical echo -n .` and `noncritical sleep 1`.

b) would just make the script a more resilient against other stray services restarting journald for whatever reason. It shouldn't happen in our controlled AMI environment but once users start customising the AMI it might happen again.
For critical commands like `mkfs` or `mount` we still would want the script to abort and fail if the command gets interrupted.

What do you think?